src/fast-path.js

```diff
@@ -583,6 +597,38 @@ function containsCallExpression(node) {
   return false;
 }
 
+function startsWithOpenCurlyBrace(node) {
+  node = getLeftMost(node);
+  switch (node.type) {
+    case "ObjectExpression":
+      return true;
+    case "MemberExpression":
+      return startsWithOpenCurlyBrace(node.object);
+    case "TaggedTemplateExpression":
+      return startsWithOpenCurlyBrace(node.tag);
+    case "CallExpression":
+      return startsWithOpenCurlyBrace(node.callee);
+    case "ConditionalExpression":
+      return startsWithOpenCurlyBrace(node.test);
+    case "UpdateExpression":
+      return !node.prefix && startsWithOpenCurlyBrace(node.argument);
+    case "BindExpression":
+      return node.object && startsWithOpenCurlyBrace(node.object);
+    case "SequenceExpression":
+      return startsWithOpenCurlyBrace(node.expressions[0])
+    default:
+      return false;
+  }
+}
+
+function getLeftMost(node) {
+  if (node.left) {
```

Comment:
Personally I'd write this function as `while (node.left) { node = node.left; } return node;`. It's mainly stylistic, but the loop is probably a little faster. Doesn't much matter though.